plugins/inputs/nfsclient/nfsclient_test.go

```diff
@@ -215,3 +215,144 @@ func TestNFSClientFileDoesNotExist(t *testing.T) {
 	nfsclient.mountstatsPath = "/does_not_exist"
 	require.Error(t, nfsclient.Gather(&acc))
 }
+
+func TestNFSClientProcessTextWithIncludeExclude(t *testing.T) {
+	// Test cases for different include/exclude scenarios
+	testCases := []struct {
+		name             string
+		includeMounts    []string
+		excludeMounts    []string
+		expectedMounts   []string
+		unexpectedMounts []string
+	}{
+		{
+			name:           "No filters",
+			includeMounts:  nil,
+			excludeMounts:  nil,
+			expectedMounts: []string{"/A", "/B"}, // All mounts should be included
+		},
+		{
+			name:             "Exclude one mount",
+			includeMounts:    nil,
+			excludeMounts:    []string{"^/A$"},
+			expectedMounts:   []string{"/B"},
+			unexpectedMounts: []string{"/A"},
+		},
+		{
+			name:             "Include one mount",
+			includeMounts:    []string{"^/A$"},
+			excludeMounts:    nil,
+			expectedMounts:   []string{"/A"},
+			unexpectedMounts: []string{"/B"},
+		},
+		{
+			name:             "Include and exclude with regex",
+			includeMounts:    []string{"^/"},
+			excludeMounts:    []string{"^/A$"},
+			expectedMounts:   []string{"/B"},
+			unexpectedMounts: []string{"/A"},
+		},
+		{
+			name:             "Exclude with prefix pattern",
+			includeMounts:    nil,
+			excludeMounts:    []string{"^/A"},
+			expectedMounts:   []string{"/B"},
+			unexpectedMounts: []string{"/A"},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var acc testutil.Accumulator
+
+			// Create NFS client with test configuration
+			nfsclient := NFSClient{
```

Comment:
How about naming the plugin instance `plugin`?

Reply:
No keep it then but using plugin everywhere would make it easier to read the test... Maybe in a separate PR...